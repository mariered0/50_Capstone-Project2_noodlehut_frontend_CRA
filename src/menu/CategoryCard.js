import React, { useState, useEffect } from "react";
import ItemCard from "./ItemCard";
import NoodleHutApi from "../api/api";
import { Container, Grid, Typography } from "@mui/material";


const CategoryCard = ({ category }) => {
  const [items, setItems] = useState([]);
  useEffect(() => {
    getItems();
  }, []);

  async function getItems() {
    const getData = await NoodleHutApi.getAllItems();
    setItems(getData);
  }


  return (
    <>
      <Typography variant="h3" align="center" color="textPrimary" gutterBottom>
        {category}
      </Typography>

      <Container sx={{ py: 8 }} maxWidth="md">
        <Grid container spacing={4}>
          {items
            ?.filter((item) => item.categoryName === `${category}`)
            .map((filteredItem) => (
              <ItemCard
                key={filteredItem.id}
                id={filteredItem.id}
                name={filteredItem.itemName}
                desc={filteredItem.itemDesc}
                price={filteredItem.itemPrice}
              />
            ))}
        </Grid>
      </Container>
    </>
  );
};

export default CategoryCard;
