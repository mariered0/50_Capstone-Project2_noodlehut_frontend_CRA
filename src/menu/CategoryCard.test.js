import React from "react";
import { render } from '@testing-library/react';
import CategoryCard from "./CategoryCard";

it("renders without crashing", () => {
    render(
        <CategoryCard />
    );
});

it("matches snapshot", () => {
	const { asFragment } = render(
        <CategoryCard />
    );
	expect(asFragment()).toMatchSnapshot();
});