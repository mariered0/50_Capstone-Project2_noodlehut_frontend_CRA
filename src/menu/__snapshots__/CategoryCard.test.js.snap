// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h3
    class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter MuiTypography-gutterBottom css-14v45o4-MuiTypography-root"
  />
  <div
    class="MuiContainer-root MuiContainer-maxWidthMd css-jcfe1v-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
    />
  </div>
</DocumentFragment>
`;
